thunder-service/service/recommendations.go

```diff
@@ -0,0 +1,164 @@
+// Copyright (c) 2020 Sorint.lab S.p.A.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+// Package service is a package that provides methods for querying data
+package service
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+
+	"github.com/ercole-io/ercole/v2/model"
+	"github.com/oracle/oci-go-sdk/v45/common"
+	"github.com/oracle/oci-go-sdk/v45/example/helpers"
+	"github.com/oracle/oci-go-sdk/v45/optimizer"
+)
+
+func (as *ThunderService) GetOCRecommendations(compartmentId string) ([]model.Recommendation, error) {
+
+	return GetOCListRecommendations(compartmentId)
+}
+
+func (as *ThunderService) GetOCRecommendationsWithCategory(compartmentId string) ([]model.RecommendationWithCategory, error) {
+
+	return GetOCListRecommendationsWithCategory(compartmentId)
+}
+
+func GetOCListRecommendations(compartmentId string) ([]model.Recommendation, error) {
+	// Create a default authentication provider that uses the DEFAULT
+	// profile in the configuration file.
+	// Refer to <see href="https://docs.cloud.oracle.com/en-us/iaas/Content/API/Concepts/sdkconfig.htm#SDK_and_CLI_Configuration_File>the public documentation</see> on how to prepare a configuration file.
+	client, err := optimizer.NewOptimizerClientWithConfigurationProvider(common.DefaultConfigProvider())
+	helpers.FatalIfError(err)
```

Comment:
Do not fatal if error, but return an error value!!!!!!!

Reply:
Updated with the correct management of the error